### PR TITLE
Documentation for Dates module, part of #31202

### PR DESCRIPTION
**Pull request by jch8ri**
updated some documentation for the Dates module

### CODE REVIEW

**Comment by fredrikekre** [comment]
What was wrong with #31945 ? It is best to update existing pull requests rather than creating new ones.

**Comment by jch8ri** [comment]
Oh sorry about that! Newbie here. Build check was failing so had to make an edit. Thanks for the feedback! @fredrikekre

File: stdlib/Dates/src/Dates.jl [hunk 1]
```diff
@@ -49,7 +49,7 @@ include("rounding.jl")
 include("io.jl")
 include("parse.jl")
 include("deprecated.jl")
-
+
```
Comment (quinnj) [inline stdlib/Dates/src/Dates.jl]: Unnecessary whitespace?

File: stdlib/Dates/src/io.jl [hunk 2]
```diff
@@ -393,9 +393,48 @@ macro dateformat_str(str)
...
+#Example
+```jldoctest
+
+julia> Dates.ISODateTimeFormat
+dateformat"yyyy-mm-ddTHH:MM:SS.s"
+
+julia> Dates.format(Dates.now(), Dates.ISODateFormat)
+"2019-04-15T17:18:05.439"
+```
+
```
Comment (quinnj) [inline stdlib/Dates/src/io.jl]: No extra blank line here

File: stdlib/Dates/src/types.jl [hunk 3]
```diff
@@ -83,6 +83,7 @@ abstract type Calendar <: AbstractTime end
 # ISOCalendar provides interpretation rules for UTInstants to civil date and time parts
 struct ISOCalendar <: Calendar end
 
+#Time Zones in ISO 8601 are represented as an offset from UTC
```
Comment (quinnj) [inline stdlib/Dates/src/types.jl]: I don't think we really need this comment as Dates doesn't really do anything timezone-related.

**Review by quinnj — CHANGES_REQUESTED** [review]
This looks mostly fine, but needs a little cleanup.

File: stdlib/Dates/src/io.jl (hunk 2 above)
Comment (fredrikekre) [reply stdlib/Dates/src/io.jl]: In addition, you need to include it in a string.

**Comment by ViralBShah** [comment]
@jch8ri Would be nice if you can implement the suggestions so we can get this in.

**Comment by ViralBShah** [comment]
Bump.

**Comment by vtjnash** [comment]
updated in #40540
